File: labs_26_04_18/ex1/Pentagon.java
package labs_26_04_18.ex1;

// Klasa dla Pięciokąta foremnego
public class Pentagon extends Figure {
    private final double side;

    public Pentagon(double side) {
        if (side <= 0)
            throw new IllegalArgumentException("Bok musi być większy od zera.");
        this.side = side;
    }

    @Override
    public double calculateArea() {
        return (Math.sqrt(5 * (5 + 2 * Math.sqrt(5))) * side * side) / 4.0;
    }

    @Override
    public double calculatePerimeter() {
        return 5 * side;
    }

    @Override
    public String getName() {
        return "Pięciokąt foremny";
    }
}

File: labs_26_04_18/ex2/src/figures/pentagon.cpp
#include "../../inc/pentagon.hpp"
#include <cmath>
#include <stdexcept>

Pentagon::Pentagon(double s) : side(s) {
  if (s <= 0)
    throw std::invalid_argument("Bok musi byc wiekszy od zera.");
}
double Pentagon::calculateArea() const {
  return (std::sqrt(5 * (5 + 2 * std::sqrt(5))) * side * side) / 4.0;
}
double Pentagon::calculatePerimeter() const { return 5 * side; }
std::string Pentagon::getName() const { return "Pieciokat foremny"; }